### ISSUE TITLE
Reintroduce a11y test after Jest is upgraded

### ISSUE THREAD

**Issue by ec1467** [mi6/ic-ui-kit#4235]
### Summary
The Stencil upgrade to v4.41.1 has caused issues with the way that jest-axe interacts with the Shadow DOM. The ic-data-table-title-bar test can't be fixed as things currently are so has been skipped. This ticket should reintroduce the test after the upgrade to Jest v30+.

### 💬 Description 
Stencil lists full support for Jest v30+ without needing to use the Stencil Test Runner. https://stenciljs.com/docs/testing/jest

Hopefully this upgrade will come along with the ability to fix this a11y test, but this isn't known. This ticket should attempt to reintroduce the test, or if it proves untenable an alternative route forward should be decided and documented. This could something small, like reducing scope to become more of a smoke test of the component, or something larger, like looking at moving a11y tests to something that runs in browser and/or is officially supported by Stencil (e.g. Playwright, Cypress).

### 🚨 Urgency (low, medium or high)
Medium. This is a known gap in test coverage and should be addressed sooner rather than later.
